sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -393,9 +394,12 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
   if (LinkOptsEnv) {
     LinkOpts = LinkOptsEnv;
   }
+  SerializedObj SpecConsts;
+  if (Prg)
+    Prg->stableSerializeSpecConstRegistry(SpecConsts);
 
   auto BuildF = [this, &M, &KSId, &Context, &Device, Prg, &CompileOpts,
```

Comment:
Just a comment. No need to address.
For further development, this function becomes rather large.
Think one's got to split it to distinct functions to improve maintainability.

Reply:
It will be done in separate patch after switch to kernel_bundle